packages/@react-native-windows/telemetry/src/utils/versionUtils.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ * @format
+ */
+
+import envinfo from 'envinfo';
+import fs from 'fs/promises';
+import path from 'path';
+import {DOMParser} from '@xmldom/xmldom';
+import xpath from 'xpath';
+
+const msbuildSelect = xpath.useNamespaces({
+  msbuild: 'http://schemas.microsoft.com/developer/msbuild/2003',
+});
+
+/**
+ * Gets the version of node being used.
+ * @returns The version of node being used.
+ */
+export async function getNodeVersion(): Promise<string | null> {
+  return process.version.slice(1);
+}
+
+/**
+ * Gets the version of npm installed, if available.
+ * @returns The version of npm installed, if available.
+ */
+export async function getNpmVersion(): Promise<string | null> {
+  try {
+    const info: any = await envinfo.helpers.getnpmInfo();
+    return info[1] as string;
+  } catch {}
+  return null;
+}
+
+/**
+ * Gets the version of yarn installed, if available.
+ * @returns The version of yarn installed, if available.
+ */
+export async function getYarnVersion(): Promise<string | null> {
+  try {
+    const info: any = await envinfo.helpers.getYarnInfo();
+    return info[1] as string;
+  } catch {}
+  return null;
+}
+
+/**
+ * Gets the latest version of Visual Studio installed, if available.
+ * @returns The latest version of Visual Studio installed, if available.
+ */
+export async function getVisualStudioVersion(): Promise<string | null> {
+  try {
+    const info: any = await envinfo.helpers.getVisualStudioInfo();
+    const versions = info[1] as string[];
+    return versions
+      .sort()
+      .slice(-1)[0]
+      .split(' ')[0];
+  } catch {}
+  return null;
+}
+
+/**
+ * Gets the version installed of the specified npm package.
+ * @param pkgName The npm package name.
+ * @returns The version installed, if available.
+ */
+export async function getVersionOfNpmPackage(
```

Comment:
How does calling `require.resolve` not find the exact place where node would be looking, starting from the place where node looks?

Reply:
It has to do with the package you start from. Because `telemetry` doesn't declare the dependency, non yarn package managers will hide it away.

In general, you always need to do resolution searching from the root of the package that declares the dependency.

Reply:
Tracking issue here: https://github.com/microsoft/react-native-windows/issues/9164